boards/components/src/sched/cooperative.rs

```diff
@@ -0,0 +1,72 @@
+//! Component for a cooperative scheduler.
+//!
+//! This provides one Component, CooperativeComponent.
+//!
+//! Usage
+//! -----
+//! (Must be final lines in main.rs)
+//! ```rust
+//! let scheduler = components::cooperative::CooperativeComponent::new(board_kernel, &PROCESSES)
+//!     .finalize(components::coop_component_helper!(NUM_PROCS));
+//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap);
```

Comment:
```suggestion
//! Note that `kernel_loop` never returns, thus this must end main.rs
//! ```rust
//! let scheduler = components::cooperative::CooperativeComponent::new(board_kernel, &PROCESSES)
//!     .finalize(components::coop_component_helper!(NUM_PROCS));
//! scheduler.kernel_loop(&imix, chip, Some(&imix.ipc), &main_cap)
```